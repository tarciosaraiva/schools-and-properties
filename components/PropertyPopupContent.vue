<template>
  <div class="popup-info container">
    <div class="row">
      <div class="column">
        <vue-flux
          :options="vfOptions"
          :images="property.media"
          :transitions="vfTransitions"
          style="width:380px"
        >
          <template v-slot:preloader>
            <flux-preloader />
          </template>
          <template v-slot:controls>
            <flux-index />
          </template>
        </vue-flux>
      </div>
    </div>
    <div class="row property">
      <span class="column price">{{ property.price }}</span>
    </div>
    <div class="row property">
      <p class="column address">
        <em>{{ property.displayableAddress }}</em>
      </p>
    </div>
    <div class="row property" v-if="property.features">
      <ul class="column">
        <li v-for="f in property.features" :key="f">
          {{ f }}
        </li>
      </ul>
    </div>
    <div class="row property">
      <span class="column">{{property.bedrooms}} rooms</span>
      <span class="column">{{property.bathrooms}} rooms</span>
      <span class="column">{{property.carspaces}} car spaces</span>
      <span class="column" v-if="property.landArea">{{property.landArea}} sqm</span>
    </div>
    <div class="row property">
      <a :href="propertyLink" target="_blank" class="column">Go to ad</a>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { VueFlux, FluxIndex, FluxPreloader } from 'vue-flux'

export default Vue.extend({
  components: {
    VueFlux,
    FluxIndex,
    FluxPreloader,
  },
  data: () => ({
    vfOptions: {
      autoplay: true,
      delay: 3000,
    },
    vfTransitions: [
      {
        name: 'slide',
        totalDuration: 300,
      },
    ],
  }),
  computed: {
    propertyLink () {
      return `https://www.domain.com.au/${this.property.listingSlug}`
    }
  },
  props: {
    property: Object,
  },
})
</script>
