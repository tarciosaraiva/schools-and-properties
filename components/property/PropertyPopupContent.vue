<template>
  <div class="popup-info container">
    <vue-flux
      :options="vfOptions"
      :images="property.media"
      :transitions="vfTransitions"
      style="width:283px"
    >
      <template #preloader>
        <flux-preloader />
      </template>
      <template #controls>
        <flux-index />
      </template>
    </vue-flux>
    <div class="property-stats">
      <p class="price">{{ property.price }}</p>
      <p class="address">
        <em>{{ property.displayableAddress }}</em>
      </p>
      <p class="division">
        <span v-show="property.bedrooms">
          <font-awesome-icon icon="bed"/>
          &nbsp;
          <span>{{property.bedrooms}}</span>
        </span>
        <span v-show="property.bathrooms">
          <font-awesome-icon icon="bath"/>
          &nbsp;
          <span>{{property.bathrooms}}</span>
        </span>
        <span v-show="property.carspaces">
          <font-awesome-icon icon="car"/>
          &nbsp;
          <span>{{property.carspaces}}</span>
        </span>
        <span v-show="property.landArea">
          <font-awesome-icon v-show="property.landArea" :icon="['far', 'square']"/>
          &nbsp;
          <span>{{property.landArea}} sqm</span>
        </span>
      </p>
      <ul v-show="property.features">
        <li v-for="f in property.features" :key="f">
          <em>{{ f }}</em>
        </li>
      </ul>
    </div>
    <div class="d-flex justify-content-around">
      <a :href="propertyLink" target="_blank" class="pl-button small full">Go to ad on domain.com.au</a>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { VueFlux, FluxIndex, FluxPreloader } from 'vue-flux'

export default Vue.extend({
  components: {
    VueFlux,
    FluxIndex,
    FluxPreloader,
  },
  props: {
    property: {
      type: Object,
      default: () => ({})
    },
  },
  data: () => ({
    vfOptions: {
      autoplay: true,
      delay: 3000,
    },
    vfTransitions: [
      {
        name: 'slide',
        totalDuration: 300,
      },
    ],
  }),
  computed: {
    propertyLink () {
      return `https://www.domain.com.au/${this.property.listingSlug}`
    }
  },
})
</script>
