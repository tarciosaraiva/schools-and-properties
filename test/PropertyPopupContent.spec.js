import { mount, createLocalVue } from '@vue/test-utils'
import { library } from '@fortawesome/fontawesome-svg-core'
import { fas } from '@fortawesome/free-solid-svg-icons'
import { far } from '@fortawesome/free-regular-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'

import PropertyPopupContent from '@/components/PropertyPopupContent.vue'

library.add(fas)
library.add(far)

const localVue = createLocalVue()
localVue.component('font-awesome-icon', FontAwesomeIcon)

describe('PropertyPopupContent', () => {
  const propsData = {
    property: {
      price: 'AUD 900K',
      media: ['https://my.image/1.png'],
      displayableAddress: '1 My Street, My Suburb',
      features: ['f1', 'f2'],
      bedrooms: 2,
      bathrooms: 2,
      carspaces: 2,
      landArea: 200,
      listingSlug: 'my-property-slug',
    }
  }

  test('is a Vue instance', () => {
    const wrapper = mount(PropertyPopupContent, { propsData, localVue })
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly with props', () => {
    const wrapper = mount(PropertyPopupContent, { propsData, localVue })
    expect(wrapper).toMatchSnapshot()
  })
})
