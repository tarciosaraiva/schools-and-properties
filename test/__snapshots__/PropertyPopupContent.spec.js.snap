// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyPopupContent renders correctly with props 1`] = `
<div class="popup-info container">
  <div class="row">
    <div class="column">
      <vue-flux-stub captions images="https://my.image/1.png" options="[object Object]" style="width: 380px;" transitions="[object Object]"></vue-flux-stub>
    </div>
  </div>
  <div class="row property">
    <span class="column price">AUD 900K</span>
  </div>
  <div class="row property">
    <p class="column address">
      <em>1 My Street, My Suburb</em>
    </p>
  </div>
  <div class="row property">
    <ul class="column">
      <li>
        f1
      </li>
      <li>
        f2
      </li>
    </ul>
  </div>
  <div class="row property">
    <span class="column">2 rooms</span>
    <span class="column">2 rooms</span>
    <span class="column">2 car spaces</span>
    <span class="column">200 sqm</span>
  </div>
  <div class="row property">
    <a class="column" href="https://www.domain.com.au/my-property-slug" target="_blank">Go to ad</a>
  </div>
</div>
`;
